src/services/DataSourceService.ts

```diff
@@ -61,4 +64,15 @@ export class DataSourceService {
             next({ statusCode: error.status || httpStatus.INTERNAL_SERVER_ERROR, message: error.message, errCode: error.code || httpStatus["500_NAME"] });
         }
     }
+    public validateDatasource = async (payload: Record<string, any>) => {
+        let datasetRecord = await ingestorService.getDatasetConfig(payload.dataset_id)
+        if (_.isEmpty(datasetRecord)) { throw constants.DATASET_NOT_FOUND }
+        else {
```

Comment:
I don't think we must handle multiple nested if and else conditions here for the validation.

1. Basic validation check (empty, undefined etc.) -> throw exception 
2. If empty -> throw exception
3. if datasource_ref invalid -> throw exception
4. If topic is invalid > throw an exception